tests/generate/test_generator.py

```diff
@@ -367,11 +400,52 @@ def sampler(biased_logits, *_):
         return torch.argmax(biased_logits, keepdims=True)
 
     generator = token_generator(MockModel(), sampler)
-    result, _, _, biased_logits = generator(None, None, None, logits_biases, None)
+    result, _, _, biased_logits = generator(
+        None, None, None, logits_biases, torch.Generator()
+    )
     assert torch.equal(result, torch.tensor(expected_result))
     assert torch.equal(biased_logits, torch.tensor(expected_biased_logits))
 
 
+@pytest.mark.parametrize(
+    "logits_biases,expected_result,expected_biased_logits",
+    [
+        ([[0, 1, 2, 3]], [[3]], [[0, 1, 2, 3]]),
+        ([[0, 1, 2]], [[2]], [[0, 1, 2, -math.inf]]),
+    ],
+)
+def test_generator_1d_cache(logits_biases, expected_result, expected_biased_logits):
+    class MockTokenizer:
+        def decode(self, _):
+            return "x"
+
+    class MockTransformerModel:
+        config = None
+
+    class MockModel:
+        tokenizer = MockTokenizer()
+        model = MockTransformerModel()
+        is_called = 0
+
+        def __call__(self, *_):
+            self.is_called += 1
```

Comment:
I know this is ugly, but I couldn't get Mocks to work mostly because of `@torch.inference_mode()`.